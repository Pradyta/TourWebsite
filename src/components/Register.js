import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

const Register = () => {
  const [name, setName] = useState("");
  const [phone, setPhone] = useState("");
  const [email, setEmail] = useState("");
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const navigate = useNavigate();

  const collectData = async () => {
    let result = await fetch("http://localhost:5000/register", {
      method: "post",
      body: JSON.stringify({ name, phone, email, username, password }),
      headers: {
        "Content-Type": "application/json",
      },
    });
    result = await result.json();

    if (result) {
      console.warn(result);
      localStorage.setItem("users", JSON.stringify(result));
      // alert("Welcome to Trip Advisor " + username);
      navigate("/");
    }
  };

  return (
    <div className="relative flex flex-col justify-center min-h-screen overflow-hidden">
      <div className="card w-full p-6 m-auto bg-white rounded-md lg:max-w-xl">
        <h1 className="text-3xl font-semibold text-center text-purple-700 underline">
          Sign Up Here
        </h1>
        <form className="mt-6">
          <div className="mb-2">
            <label
              htmlFor="name"
              className="block text-sm font-semibold text-gray-800"
            >
              Name
            </label>
            <input
              type="text"
              value={name}
              onChange={(e) => setName(e.target.value)}
              className="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            />
          </div>

          <div className="mb-2">
            <label
              htmlFor="contact"
              className="block text-sm font-semibold text-gray-800"
            >
              Contact
            </label>
            <input
              type="number"
              value={phone}
              onChange={(e) => setPhone(e.target.value)}
              className="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            />
          </div>

          <div className="mb-2">
            <label
              htmlFor="email"
              className="block text-sm font-semibold text-gray-800"
            >
              Email
            </label>
            <input
              type="text"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              className="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            />
          </div>

          <div className="mb-2">
            <label
              htmlFor="newname"
              className="block text-sm font-semibold text-gray-800"
            >
              User Name
            </label>
            <input
              type="text"
              value={username}
              onChange={(e) => setUsername(e.target.value)}
              className="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            />
          </div>

          <div className="mb-2">
            <label
              htmlFor="newpassword"
              className="block text-sm font-semibold text-gray-800"
            >
              Password
            </label>
            <input
             type="password"
            placeholder="Password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              className="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            />
          </div>

          <div className="mt-6">
            <button
              type="button"
              className="w-full px-4 py-2 tracking-wide text-white transition-colors duration-200 transform bg-purple-700 rounded-md hover:bg-purple-600 focus:outline-none focus:bg-purple-600"
              onClick={collectData}
            >
              Sign Up
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default Register;
