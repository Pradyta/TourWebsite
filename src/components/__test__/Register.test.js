import { render, screen } from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import Register from '../Register';

describe('Register Component', () => {
    test('should render Register compoennt', () => {
    const {asFragment} = render( 
    <Router>
        <Register />,
    </Router>
    );
    expect(asFragment(<Register/>)).toMatchSnapshot();
    });


    test("render the Register form submit button on the screen", async () => {
      render(<Router>
        <Register />,
    </Router>);
      const buttonList = await screen.findAllByRole("button");
      expect(buttonList).toHaveLength(1);
    });

    test("passport input should have type password ", () => {
        render(<Router>
        <Register />,
    </Router>);
        const password = screen.getByPlaceholderText("Password");
        expect(password).toHaveAttribute("type", "password");
      });
});
