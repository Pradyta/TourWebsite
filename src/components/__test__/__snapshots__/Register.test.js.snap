// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component should render Register compoennt 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col justify-center min-h-screen overflow-hidden"
  >
    <div
      class="w-full p-6 m-auto bg-white rounded-md shadow-md lg:max-w-xl"
    >
      <h1
        class="text-3xl font-semibold text-center text-purple-700 underline"
      >
        Sign Up Here
      </h1>
      <form
        class="mt-6"
      >
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="name"
          >
            Name
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="contact"
          >
            Contact
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            type="number"
            value=""
          />
        </div>
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="email"
          >
            Email
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="newname"
          >
            User Name
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="newpassword"
          >
            Password
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div
          class="mt-6"
        >
          <button
            class="w-full px-4 py-2 tracking-wide text-white transition-colors duration-200 transform bg-purple-700 rounded-md hover:bg-purple-600 focus:outline-none focus:bg-purple-600"
            type="button"
          >
            Sign Up
          </button>
        </div>
      </form>
    </div>
  </div>
  ,
</DocumentFragment>
`;
